server/src/test/java/org/elasticsearch/index/store/StoreTests.java

```diff
@@ -1187,4 +1187,14 @@ public void testVersionIsIncludedInBootstrapCommit() throws IOException {
             assertThat(segmentInfos.getUserData().get(Engine.ES_VERSION), is(equalTo(org.elasticsearch.Version.CURRENT.toString())));
         }
     }
+
+    public void testReadMetadataSnapshotReturnsEmptyWhenIndexIsCorrupted() throws IOException {
+        var shardId = new ShardId("index", "_na_", 1);
+        var dir = createTempDir();
+        try (Store store = new Store(shardId, INDEX_SETTINGS, new NIOFSDirectory(dir), new DummyShardLock(shardId))) {
+            store.markStoreCorrupted(new IOException("test exception"));
```

Comment:
Suggest this otherwise the store is empty anyway:

```suggestion
            store.createEmpty();
            store.markStoreCorrupted(new IOException("test exception"));
```